pkg/secrets/handlers/secrets_handler_k8s_test.go

```diff
@@ -29,4 +30,35 @@ func TestSecretsHandlerK8sUseCase(t *testing.T) {
 			So(err.Error(), ShouldEqual, "error map should not be empty")
 		})
 	})
+
+	Convey("updateK8sSecretsMapWithConjurSecrets", t, func () {
+		secret := []byte{'s', 'u', 'p', 'e', 'r'}
+		conjurSecrets := make(map[string][]byte)
+		conjurSecrets["account:variable:allowed/username"] = secret
+
+		newDataEntriesMap := make(map[string][]byte)
+		newDataEntriesMap["username"] = []byte("allowed/username")
+
+		k8sSecretsMap := make(map[string]map[string][]byte)
+		k8sSecretsMap["mysecret"] = newDataEntriesMap
+
+		pathMap := make(map[string]string)
+		pathMap["allowed/username"] = "mysecret:username"
+
+		k8sSecretsStruct := k8s.K8sSecretsMap{k8sSecretsMap, pathMap}
+		err := updateK8sSecretsMapWithConjurSecrets(&k8sSecretsStruct, conjurSecrets)
+
+		eq := reflect.DeepEqual(k8sSecretsStruct.K8sSecrets["mysecret"]["username"], secret)
+		Convey("Returns true if K8sSecretsMap's variable ID is replaced with corresponding secret value", func() {
+			So(err, ShouldEqual, nil)
+			So(eq, ShouldEqual, true)
+		})
+
+		Convey("Returns true if Conjur secret byte array is cleared from memory", func() {
+			// secret value should be cleared from memory after use
+			empty := make([]byte, len(secret))
+			eq = reflect.DeepEqual(conjurSecrets["account:variable:allowed/username"], empty)
+			So(eq, ShouldEqual, true)
```

Comment:
does this actually work? did you change the code where we clear the secret from memory?